docs/proposals/valkey.md

```diff
@@ -0,0 +1,82 @@
+---
+title: Replace Redis with ValKey
+authors:
+  - "@hairmare"
+sponsors:
+  - TBD        # List all interested parties here.
+reviewers:
+  - "@alexmt"
+  - TBD
+approvers:
+  - "@alexmt"
+  - TBD
+
+creation-date: 2024-04-19
+last-updated: 2024-04-19
+---
+
+# Replace ValKey with Redis
+
+## Summary
+
+Redis Inc announced that they are changing Redis'  license to the source available BUSL license [here](https://redis.com/blog/redis-adopts-dual-source-available-licensing/) which is in violation of [CNCF policy](https://github.com/cncf/foundation/blob/main/allowed-third-party-license-policy.md) and does not align with the Argo Projects values.
+
+The BSD-3-Clause licensed Redis OSS code has now been forked as [ValKey](https://valkey.io/) in an effort hosted by the Linux Foundation.
+
+## Motivation
+
+Taking swift action when this kind of change happen in our supply chain is at the core of upholding the open source values of the Argo Project.
+
+### Goals
+
+* Redis is replaced with ValKey in the Argo CD project.
+* This afford downstream distributions of Argo CD the security the security that they can follow Argo CD's lead
+
+### Non-Goals
+
+Replacing the Redis wire-protocol or any fundamental architecture changes are out of scope for this proposal.
+It is assumed that ValKey will take over Redis' leading position and we are merely switch upstream with the community.
+
+## Proposal
+
+Given the nature of this change, fundamentally it should be as simple as replacing the container images in [manifests/base/redis](https://github.com/argoproj/argo-cd/tree/3a46e8c1c7dc20911cb5d87ade8ced26c766e273/manifests/base/redis).
```

Comment:
Also see [manifests/ha/base/redis-ha](https://github.com/argoproj/argo-cd/tree/master/manifests/ha/base/redis-ha) and [manifests/ha/base/redis-ha/chart/requirements.yaml](https://github.com/argoproj/argo-cd/blob/master/manifests/ha/base/redis-ha/chart/requirements.yaml)

For HA install of ArgoCD, the redis-ha configuration is provided by a 3rd party helm chart:

https://github.com/DandyDeveloper/charts/tree/master/charts/redis-ha